nautobot/extras/context_managers.py

```diff
@@ -202,9 +203,13 @@ def web_request_context(
         with change_logging(change_context):
             yield request
     finally:
+        # For anything except a JOB_HOOK context, we might be triggering job hooks, so make sure they're up to date
+        if context != ObjectChangeEventContextChoices.CONTEXT_JOB_HOOK:
+            get_jobs(reload=True)
         # enqueue jobhooks and webhooks, use change_context.change_id in case change_id was not supplied
         for object_change in ObjectChange.objects.filter(request_id=change_context.change_id).iterator():
-            enqueue_job_hooks(object_change)
+            if context != ObjectChangeEventContextChoices.CONTEXT_JOB_HOOK:
```

Comment:
Not strictly necessary since the first thing `enqueue_job_hooks` does is check whether `object_change.change_context` is `JOB_HOOK` and returns immediately if so, but I felt it was clearer to add.